### PR TITLE
Add structured content output to get_lineage tool

### PR DESCRIPTION
**Pull request by jgiuffrida**
## Summary
- Return a `LineageGraph` model via `structuredContent` on `CallToolResult`, enabling MCP hosts to consume lineage as typed nodes and edges
- Raw JSON nodes are still returned in `content` for backwards compatibility
- Add unit tests for the structured output transformation (graph shape, edge filtering, empty results, missing parentIds)

## Test plan
- [x] `task check` passes (ruff, mypy, all linters)
- [x] `task test:unit` passes (452 tests including 7 new)
- [ ] Manual verification with MCP Inspector

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/dbt_mcp/discovery/tools.py [hunk 1]
```diff
@@ -233,22 +236,63 @@ async def get_model_performance(
...
 )
 async def get_lineage(
     context: DiscoveryToolContext,
     unique_id: str = UNIQUE_ID_REQUIRED_FIELD,
     types: list[LineageResourceType] | None = TYPES_FIELD,
     depth: int = DEPTH_FIELD,
-) -> list[dict]:
-    return await context.lineage_fetcher.fetch_lineage(
+) -> Annotated[CallToolResult, LineageGraph]:
+    nodes = await context.lineage_fetcher.fetch_lineage(
```
Comment (DevonFulcher) [inline src/dbt_mcp/discovery/tools.py:272]: Perhaps `fetch_lineage` should return the pydantic model? Less dicts and more typed objects in general are better.
